e2e/scenario_helpers_test.go

```diff
@@ -153,3 +159,73 @@ func getExpectedPackageVersions(packageName, distro, release string) []string {
 	}
 	return expectedVersions
 }
+
+func getCustomScriptExtensionStatus(ctx context.Context, t *testing.T, resourceGroupName, vmssName string) error {
+	pager := config.Azure.VMSSVM.NewListPager(resourceGroupName, vmssName, nil)
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get VMSS instances: %v", err)
+		}
+
+		for _, vmInstance := range page.Value {
+			instanceViewResp, err := config.Azure.VMSSVM.GetInstanceView(ctx, resourceGroupName, vmssName, *vmInstance.InstanceID, nil)
+			if err != nil {
+				t.Logf("failed to get instance view for VM %s: %v", *vmInstance.InstanceID, err)
+				continue
```

Comment:
Is there a reason to continue instead of returning the error?

Reply:
thought it might still be worth to check the other vms if multiple in the scale set but in this case of ab e2es we only have 1 instance anyway. fixed to return error